VocaDbWeb.Core/Code/VocaDbPage.cs

```diff
@@ -62,12 +63,12 @@ public string ToJS(int? val)
 
 		public IHtmlContent ToJS(string str)
 		{
-			return new HtmlString(JsonHelpers.Serialize(str));
+			return new HtmlString($@"JSON.parse(""{HttpUtility.JavaScriptStringEncode(JsonHelpers.Serialize(str))}"")");

```

Comment:
You should extract this into a helper method of its own, possibly static, so it doesn't have to be repeated everywhere.